src/app/service/categories/ServiceCategories.jsx

```diff
@@ -21,12 +22,34 @@ const LoadingView = () => (
 );
 
 class ServiceCategories extends Component {
-  state = { active: -1, selected: {} };
+  state = {
+    active: -1,
+    selected: {},
+    currentCategories: {},
+    isLoading: false,
+  };
 
   componentWillMount() {
     if (!this.props.taxonomy) {
       this.props.getTaxonomy();
     }
+    if (!this.props.location) {
+      const { locationId } = this.props.match.params;
+      this.props.getLocation(locationId);
+    }
+  }
+
+  componentWillReceiveProps(nextProps, prevState) {
```

Comment:
Shouldn't matter too much here, but I think with React, copying props to unchanging state is often considered an anti-pattern. Was the reason for it here concern about the performance impact of needing to map the location data to categories every time something (unrelated) changes in the state?
If so, I'm not sure if that isn't somewhat premature optimization - we're really just talking about iterating over a small handful of objects.
And even if we were very concerned about the performance of this, there's always the more foundational solutions, like using a memoized selector (e.g. with `reselect`) to map the prop to begin with in mapStateToProps.

Reply:
Was just finding a place to update the current categories. I guess this belongs in the reducer instead.